app/src/main/java/com/github/warnastrophy/core/di/AppDependencies.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.warnastrophy.core.di
+
+import android.content.Context
+import com.github.warnastrophy.core.data.repository.ContactRepositoryProvider
+import com.github.warnastrophy.core.data.repository.HazardRepositoryProvider
+import com.github.warnastrophy.core.model.ErrorHandler
+import com.github.warnastrophy.core.model.GpsService
+import com.github.warnastrophy.core.model.GpsServiceFactory
+import com.github.warnastrophy.core.model.HazardsService
+import com.github.warnastrophy.core.model.HazardsServiceFactory
+import com.github.warnastrophy.core.model.PermissionManager
+import com.github.warnastrophy.core.model.PermissionManagerInterface
+import com.google.android.gms.location.LocationServices
+
+object AppDependencies {
+  private lateinit var _errorHandler: ErrorHandler
+  val errorHandler
+    get() = _errorHandler
+
+  private lateinit var _gpsService: GpsService
+  val gpsService
+    get() = _gpsService
+
+  private lateinit var _hazardsService: HazardsService
+  val hazardsService
+    get() = _hazardsService
+
+  private lateinit var _permissionManager: PermissionManagerInterface
+  val permissionManager
+    get() = _permissionManager
```

Comment:
```suggestion
  private var _errorHandler: ErrorHandler? = null
  val errorHandler
    get() =
      _errorHandler
        ?: throw IllegalStateException("ErrorHandler not initialized. Did you call AppDependencies.init()?")

  private var _gpsService: GpsService? = null
  val gpsService
    get() =
      _gpsService
        ?: throw IllegalStateException("GpsService not initialized. Did you call AppDependencies.init()?")

  private var _hazardsService: HazardsService? = null
  val hazardsService
    get() =
      _hazardsService
        ?: throw IllegalStateException("HazardsService not initialized. Did you call AppDependencies.init()?")

  private var _permissionManager: PermissionManagerInterface? = null
  val permissionManager
    get() =
      _permissionManager
        ?: throw IllegalStateException("PermissionManager not initialized. Did you call AppDependencies.init()?")
```

AppDependencies uses lateinit backing fields, so accessing them before init throws UninitializedPropertyAccessException with little context. This is why I suggest to check the variables.

Reply:
Done